src/app/dataset-block/metadata-block/components/event-details/builder.events.ts

```diff
@@ -170,9 +182,105 @@ class SetPollingSourceSectionBuilder extends EventSectionBuilder<SetPollingSourc
     }
 }
 
+class SetTransformSectionBuilder extends EventSectionBuilder<SetTransform> {
+    private readonly UNSUPPORTED_ROW_DESCRIPTOR: EventRowDescriptor = {
+        ...SET_TRANSFORM_SOURCE_DESCRIPTORS[`SetTransform.UnsupportedKey`],
+    };
+
+    public buildEventSections(event: SetTransform): EventSection[] {
+        const result: EventSection[] = [];
+        Object.entries(event).forEach(([section, data]) => {
+            if (section !== "__typename") {
+                switch (section) {
+                    case SetTransformSourceSection.TRANSFORM: {
+                        result.push({
+                            title: section,
+                            rows: this.buildEventRows(data as TransformSql),
+                        });
+
+                        break;
+                    }
+                    case SetTransformSourceSection.INPUTS: {
+                        const numInputsParts = event.inputs.length;
+                        (data as TransformInput[]).forEach((item, index) => {
+                            result.push({
+                                title:
+                                    section +
+                                    (numInputsParts > 1 ? `#${index + 1}` : ""),
+                                rows: this.buildEventRows(
+                                    item.dataset as unknown as TransformInput,
+                                ),
+                            });
+                        });
+                        break;
+                    }
+                    default: {
+                        result.push({ title: section, rows: [] });
+                    }
+                }
+            }
+        });
+
+        return result;
+    }
+
+    private buildEventRows(
```

Comment:
This method should be generalized and placed in the base class

Reply:
Moved to abstract class